src/lib/comms/include/sol-lwm2m.h

```diff
@@ -733,10 +755,34 @@ void sol_lwm2m_resource_clear(struct sol_lwm2m_resource *resource);
  * before calling this function.
  */
 int sol_lwm2m_resource_init(struct sol_lwm2m_resource *resource,
-    uint16_t id, uint16_t resource_len,
+    uint16_t id, enum sol_lwm2m_resource_type type, uint16_t resource_len,
     enum sol_lwm2m_resource_data_type data_type, ...);
 
 /**
+ * @brief Initializes a LWM2M resource of type multiple using two @c sol_vector.
+ *
+ * This function makes it easier to init a LWM2M resource of type multiple,
+ * when we don't know beforehand the amount of Resource Instances desired.
+ * The last argument is a @c sol_vector holding elements of the relevant
+ * resource data type as per the table in @see sol_lwm2m_resource_init().
+ *
+ * @param resource The resource to be initialized.
+ * @param id The resource id.
+ * @param data_type The resource data type.
+ * @param res_inst_ids The list of Resource Instance IDs, in the same order as their relevant values in @c res_inst_values
+ * @param res_inst_values The list of Resource Instance values, in the same order as their relevant ids in @c res_inst_ids, respecting the table according to the resource data type.
+ * @return 0 on success, negative errno on error.
+ * @see sol_lwm2m_resource_clear()
+ * @see sol_lwm2m_resource_init()
+ *
+ * @note The LWM2M resource api_version must be set,
+ * before calling this function.
+ */
+int sol_lwm2m_resource_init_vector(struct sol_lwm2m_resource *resource,
```

Comment:
Mixed feelings in here.

An alternative I thought of would be to expose the type `struct sol_lwm2m_resource_data` (already used inside `sol_lwm2m_resource`), and the user could provide a single `sol_vector` with elements from this type. I'm not sure about how better that would be, though...